pandas/core/internals.py

```diff
@@ -4435,14 +4435,6 @@ def _interleaved_dtype(blocks):
     for x in blocks:
         counts[type(x)].append(x)
 
-    def _lcd_dtype(l):
-        """ find the lowest dtype that can accomodate the given types """
-        m = l[0].dtype
-        for x in l[1:]:
```

Comment:
what I meant was can start off by simply moving this (your version or new one with ``np.find_common_type`` to ``pandas.types.cast`` (and if we have specific tests move similarly; if not, ideally add some). we can add the pandas specific functionaility later.